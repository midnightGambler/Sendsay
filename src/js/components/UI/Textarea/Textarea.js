import React from "react";

const Textarea = ({ label, error, ...rest }) => (
  <div className="textarea textarea-mb">
    <label className="textarea__label">{label}</label>
    <textarea className="textarea__textarea" {...rest} />
    {error && <span className='input-error'>{error}</span>}
  </div>
);

export default Textarea;
