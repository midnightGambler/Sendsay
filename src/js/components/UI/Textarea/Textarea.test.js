import React from "react";
import { shallow } from "enzyme";
import Textarea from "./Textarea";

it("should render component properly", () => {
  const textarea = shallow(<Textarea label="test" />);

  expect(textarea).toMatchSnapshot();
});

it("should render component with error field", () => {
  const textarea = shallow(<Textarea error="test error" label="test" />);

  expect(textarea).toMatchSnapshot();
});

it("should test component onChange event", () => {
  const mockedCallback = jest.fn();
  const textarea = shallow(<Textarea onChange={mockedCallback} label="test" />);
  textarea.find("textarea").simulate("change");
  expect(mockedCallback.mock.calls.length).toBe(1);
});
