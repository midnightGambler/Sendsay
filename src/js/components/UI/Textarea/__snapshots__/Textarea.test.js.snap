// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component properly 1`] = `
<div
  className="textarea textarea-mb"
>
  <label
    className="textarea__label"
  >
    test
  </label>
  <textarea
    className="textarea__textarea"
  />
</div>
`;

exports[`should render component with error field 1`] = `
<div
  className="textarea textarea-mb"
>
  <label
    className="textarea__label"
  >
    test
  </label>
  <textarea
    className="textarea__textarea"
  />
  <span
    className="input-error"
  >
    test error
  </span>
</div>
`;
